vault/resource_kv_secret_v2.go

```diff
@@ -184,13 +254,62 @@ func kvSecretV2Read(_ context.Context, d *schema.ResourceData, meta interface{})
 				if err := d.Set(consts.FieldMetadata, serializeDataMapToString(v)); err != nil {
 					return diag.FromErr(err)
 				}
+
+				// Read & Set custom metadata
+				if _, ok := v[consts.FieldCustomMetadata]; ok {
+					cm, err := readKVV2Metadata(d, client)
+					if err != nil {
+						return diag.FromErr(err)
+					}
+
+					if err := d.Set(consts.FieldCustomMetadata, []interface{}{cm}); err != nil {
+						return diag.FromErr(err)
+					}
+				}
 			}
 		}
+
 	}
 
 	return nil
 }
 
+func readKVV2Metadata(d *schema.ResourceData, client *api.Client) (map[string]interface{}, error) {
+	path := strings.Replace(d.Id(), consts.FieldData, consts.FieldMetadata, 1)
+
+	log.Printf("[DEBUG] Reading metadata for KVV2 secret at %s", path)
+	resp, err := client.Logical().Read(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp == nil {
+		log.Printf("[DEBUG] no metadata found for secret")
+		return nil, nil
+	}
+
+	data := map[string]interface{}{}
+
+	for vaultKey, tfKey := range metadataFields {
+		if val, ok := resp.Data[vaultKey]; ok {
+			// the delete_version_after field is written to
+			// Vault as an integer but is returned as a string
+			// of the format "3h12m10s"
+			if vaultKey == consts.FieldDeleteVersionAfter {
+				t, err := time.ParseDuration(val.(string))
+				if err != nil {
+					return nil, fmt.Errorf("error parsing duration, err=%s", err)
+				}
+				data[tfKey] = t.Seconds()
```

Comment:
nit: you can probably set `val = t.Seconds()` here, then you can drop the `else{}`. Makes things a bit easier to read.